ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1326,6 +1369,9 @@ void onFillComplete(boolean updateMetric) {
         routerMetrics.chunkFillTimeMs.update(time.milliseconds() - chunkFreeAtMs);
       }
 
+      // Attempt to apply compression on data chunk.
+      compressChunk();
```

Comment:
Would we eventually make compression and decompression asynchronous as well?

Reply:
No, compression and decompression do not need to be asynchronous because we are already running in a worker thread, not Netty IO thread.  Make it async will increase complexity and slow down operation.